### PR TITLE
feat(devtools-extension): Add extension version

### PR DESCRIPTION
**Pull request by WayneFerrao**
[AB#4995](https://dev.azure.com/fluidframework/235294da-091d-4c29-84fc-cdfc3d90890b/_workitems/edit/4995)

## Description

This PR uses the chrome runtime API to fetch and add the devtools extension version to the extension generated usage telemetry logs

### CODE REVIEW

**Review by alexvy86 — COMMENTED** [review]
@WayneFerrao first of all good job with the PR, it's pretty solid. I will admit a bit of lack of foresight around this task... I think we don't want to have devtools-view care about this new property at all, precisely because it's something that would only be available in the extension. I think where we need to add the new property is in the "external" logger that the extension will provide to actually send logs to our telemetry collection endpoint. It means the extension version wouldn't show up in the logs when we output them to console (even in the extension's console), we'd only see it in Aria/Kusto, but I think that's correct.

It would mean that the changes for this task don't have a place to be applied yet, though. We could apply them to the [branch with the experimental logger using the internal packages](https://github.com/microsoft/FluidFramework/pull/16351), but not sure it's very valuable since we're probably not going with that approach. Once we have a final logger implementation for the extension using the public packages (task for me for this sprint), then we could make this change there, so the external logger adds this property to the logs that flow through it.

@Josmithr curious if you have any thoughts here.

I'd say let's move the task to in-review, because this PR shows pretty clearly how to get the value. If I don't get that final implementation done before you leave on vacation @WayneFerrao , I can take over applying this change on the logger.

@rohandubal FYI (and in case you have thoughts).

**Comment by alexvy86** [comment]
We could also take an approach where we take a property bag from "outside" and add them to all the events when we generate them (basically what the PR does with `versionExtension` as a prop for `DevtoolsPanel` but taking an arbitrary set of key-value pairs). Since devtools-view is not intended as a library for external consumers to import, I think that might be alright... but only having the external logger add properties as things flow through it still feels cleaner to me.

File: packages/tools/devtools/devtools-view/package.json [hunk 1]
```diff
@@ -50,6 +50,7 @@
 		"@fluidframework/driver-definitions": "workspace:~",
 		"@fluidframework/protocol-definitions": "^1.1.0",
 		"@fluidframework/telemetry-utils": "workspace:~",
+		"@types/chrome": "0.0.232",
```
Comment (Josmithr) [inline packages/tools/devtools/devtools-view/package.json:53]: Is this needed for something specific? Our React view really shouldn't need to know anything about chrome's APIs if we can help it.

**Comment by Josmithr** [comment]
> It would mean that the changes for this task don't have a place to be applied yet, though. We could apply them to the [branch with the experimental logger using the internal packages](https://github.com/microsoft/FluidFramework/pull/16351), but not sure it's very valuable since we're probably not going with that approach. Once we have a final logger implementation for the extension using the public packages (task for me for this sprint), then we could make this change there, so the external logger adds this property to the logs that flow through it.
> 
> @Josmithr curious if you have any thoughts here.

Yeah, that sounds right to me.

> We could also take an approach where we take a property bag from "outside" and add them to all the events when we generate them (basically what the PR does with `versionExtension` as a prop for `DevtoolsPanel` but taking an arbitrary set of key-value pairs). Since devtools-view is not intended as a library for external consumers to import, I think that might be alright... but only having the external logger add properties as things flow through it still feels cleaner to me.

I was going to suggest basically the same thing. Something like "environmentData", which is extra data logged alongside telemetry to note the environment in which the view is being run. The example app could optionally specify some things there, and the extension can specify the extension version and maybe even browser type / version details (if we're allowed to log that).

Also, while the view package is not currently published or extended for external use, I would like to continue treating it as an isolated library that _could_ be made public in the future. So, we should be careful about adding adding extension-specific details or documentation in the view package, IMO.

**Comment by alexvy86** [comment]
This kind of ties into the discussion we've had for Fluid logging as well. I do lean towards having enrichment of logs be entirely a responsibility of the sinks (in this case the external logger) instead of giving them hooks to alter the messages when we generate them. It's also less work, so I'll suggest we go that route, and integrate this change into the PR with the final external logger implementation.

**Comment by alexvy86** [comment]
Closing in favor of https://github.com/microsoft/FluidFramework/pull/16969

**Pull request closed without merging**